internal/service/watcher/certificate/renewal/gcm/renewal_service.go

```diff
@@ -42,6 +46,32 @@ func (s *Service) Renew(ctx context.Context, name string) error {
 	return nil
 }
 
+// SkrSecretNeedsRenewal check if the SKR Secret needs to be renewed.
+// Renewal is required if the gateway certificate secret is newer than the SKR certificate secret.
+func (s *Service) SkrSecretNeedsRenewal(gatewaySecret, skrSecret *apicorev1.Secret) bool {
+	gwSecretLastModifiedAtValue, ok := gatewaySecret.Annotations[shared.LastModifiedAtAnnotation]
+	if !ok {
+		return true
+	}
+
+	gwSecretLastModifiedAt, err := time.Parse(time.RFC3339, gwSecretLastModifiedAtValue)
+	if err != nil {
+		return true
+	}
+
+	lastRequestedAtValue, ok := skrSecret.Annotations[shared.GCMSecretAnnotation]
+	if !ok {
+		return true
+	}
+
+	lastRequestedAtValueTime, err := time.Parse(time.RFC3339, lastRequestedAtValue)
```

Comment:
Maybe rename to `lastRequestedAt` for consitency with `gwSecretLastModifiedAt`? Otherwise, I would wonder why we have `ValueTime` suffix for one but not the other.